handlers/issue.go

```diff
@@ -1,71 +1,80 @@
 package handlers
 
 import (
+	"context"
 	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
 	"regexp"
 	"strings"
 
 	"github.com/golang/glog"
 	"github.com/google/go-github/github"
 )
 
+const (
+	Assign   = "/assign"
+	Unassign = "/unassign"
+)
+
 type GithubIssue github.Issue
 
 func (s *Server) handleIssueEvent(body []byte) {
-	glog.Infof("Received an Issue Event")
+	glog.Info("Received an Issue Event")
 
 }
 
-func (s *Server) handleIssueCommentEvent(body []byte) {
-	glog.Infof("Received an IssueComment Event")
+//function to handle issue comments
+func (s *Server) handleIssueCommentEvent(body []byte, client *github.Client) {
+	var commentEvent github.IssueCommentEvent
 
-	var prc github.IssueCommentEvent
-	err := json.Unmarshal(body, &prc)
+	err := json.Unmarshal(body, &commentEvent)
 	if err != nil {
 		glog.Errorf("fail to unmarshal: %v", err)
 	}
-	glog.Infof("prc: %v", prc)
-	comment := *prc.Comment.Body
+	//Get the owner info from the req body, and use this owner for the all api's
+	owner := *commentEvent.Repo.Owner.Login
+	ctx := context.Background()
+	//extract the user from login to validate whether user is collaborator or not
+	user := commentEvent.Sender.Login
+	comment := *commentEvent.Comment.Body
+	//split the assignees and operation to be performed.
+	substrings := strings.Split(comment, "@")
 
-	// https://github.com/islinwb/test/pull/1
-	prID := strings.SplitAfter(prc.Issue.PullRequestLinks.GetHTMLURL(), "github.com/")[1]
-	// https://github.com/islinwb/test/pull/1.patch
-	// From <commit ID> MON ...
-	patchURL := prc.Issue.PullRequestLinks.GetPatchURL()
-	resp, err := http.Get(patchURL)
+	assign, _, err := client.Repositories.IsCollaborator(ctx, owner, *commentEvent.Repo.Name, *user)
```

Comment:
A quick question, do we only assign to people who is the collaborator of the repo?

Seems unreasonable to me.

Reply:
@m1093782566 , No, it will assign even if the user is not a collaborator. it will return false when the assigned user is not a github user.

Reply:
Can you please explain the usage of `IsCollaborator()`? I just feel wired that we use `IsCollaborator()` to identify github user.

Reply:
IsCollaborator checks whether the specified GitHub user has collaborator access to the given repo. And IsCollaborator will return false if the user is not a collaborator OR the user
is not a GitHub user.
GitHub API docs: https://developer.github.com/v3/repos/collaborators/#get

Reply:
In general, If suppose I comment on the repo /assign Dujun .... This will check if sids-b is collaborator of the repo to prevent misuse of bot by not allowing anyone to assign an issue or PR to a person.
Correct me if i am wrong and thoughts regarding the same ??

Reply:
I think I need to clarify the requirement. In Kubernetes, we can only assign PR/issue to people who are outside collaborator OR Kubernetes org member. In kubeedge, I think we don't need to follow this way for now as we have no rules on managing kubeedge memberships.